eclipse-collections-code-generator/src/main/resources/test/list/mutable/boxedMutableArrayListTest.stg

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Test
+    public void remove()
+    {
+        Assert.assertEquals(<(literal.(type))("2")>, (<type>) this.classUnderTest().remove(1)<wideDelta.(type)>);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void removeIndexOutOfBounds()
+    {
+        this.classUnderTest().remove(10);
+    }
+
+    @Test
+    public void indexOf()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        Assert.assertEquals(1, list.indexOf(<(literal.(type))("2")>));
+        Assert.assertEquals(-1, list.indexOf(<(literal.(type))("5")>));
+    }
+
+    @Test
+    public void lastIndexOf()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        list.add(<(literal.(type))("1")>);
+        Assert.assertEquals(3, list.lastIndexOf(<(literal.(type))("1")>));
+        Assert.assertEquals(1, list.lastIndexOf(<(literal.(type))("2")>));
+        Assert.assertEquals(-1, list.indexOf(<(literal.(type))("5")>));
+    }
+
+    @Test
+    public void set()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        Assert.assertEquals(<(literal.(type))("2")>, (<type>) list.set(1, <(literal.(type))("5")>)<wideDelta.(type)>);
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "5", "3"]:(literal.(type))(); separator=", ">)), list);
+
+        Assert.assertEquals(<(literal.(type))("5")>, (<type>) list.set(1, <(literal.(type))("7")>)<wideDelta.(type)>);
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "7", "3"]:(literal.(type))(); separator=", ">)), list);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void setIndexOutOfBounds()
+    {
+        this.classUnderTest().set(5, <(literal.(type))("5")>);
+    }
+
+    @Test
+    public void get()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        Assert.assertEquals(<(literal.(type))("1")>, (<type>) list.get(0)<wideDelta.(type)>);
+        Assert.assertEquals(<(literal.(type))("2")>, (<type>) list.get(1)<wideDelta.(type)>);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void getIndexOutOfBounds()
+    {
+        this.classUnderTest().get(5);
+    }
+
+    @Test
+    public void addAll()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        list.addAll(1, Lists.mutable.of(<(literal.(type))("7")>, <(literal.(type))("9")>));
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "7", "9", "2", "3"]:(literal.(type))(); separator=", ">)), list);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void addAllIndexOutOfBounds()
+    {
+        this.classUnderTest().addAll(5, Lists.mutable.of(<(literal.(type))("7")>, <(literal.(type))("9")>));
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void subList()
+    {
+        this.classUnderTest().subList(1, 3);
+    }
+}
+
+>>
```

Comment:
Since this is a view, I think it would be useful to have at least a few assertions where we mutate the original (the primitive list) and see the mutations through the view.

Reply:
Added a new test. Thanks.